src/MarketV2.sol

```diff
@@ -0,0 +1,679 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+import "src/interfaces/IERC20.sol";
+
+// Caution. We assume all failed transfers cause reverts and ignore the returned bool.
+interface IOracle {
+    function getPrice(address,uint) external returns (uint);
+    function viewPrice(address,uint) external view returns (uint);
+}
+
+interface IEscrow {
+    function initialize(IERC20 _token, address beneficiary) external;
+    function onDeposit() external;
+    function onDepositCallBack() external returns(uint);
+    function pay(address recipient, uint amount) external;
+    function balance() external view returns (uint);
+}
+
+interface IDolaBorrowingRights {
+    function onBorrow(address user, uint additionalDebt) external;
+    function onRepay(address user, uint repaidDebt) external;
+    function onForceReplenish(address user, address replenisher, uint amount, uint replenisherReward) external;
+    function balanceOf(address user) external view returns (uint);
+    function deficitOf(address user) external view returns (uint);
+    function replenishmentPriceBps() external view returns (uint);
+}
+
+interface IBorrowController {
+    function borrowAllowed(address msgSender, address borrower, uint amount) external returns (bool);
+    function onRepay(uint amount) external;
+}
+
+interface IDebtManager {
+	function debt(address borrower) external view returns(uint);
+	function totalDebt() external view returns(uint);
+	function dbrDeficit(address borrower) external view returns(uint);
```

Comment:
Debt managers shouldn't have to deal with DBR and therefore this function shouldn't be on the interface. Maybe it can be replaced with a boolean `inDeficit(address) returns (bool)` function which would be universal whether using DBR or not.